Add static labels support for the AWS Firehose Logs (#740)
grafana/alloy: docs/sources/reference/components/loki.source.awsfirehose.md

The component will start an HTTP server on the configured port and address with the following endpoints:

- `/awsfirehose/api/v1/push` - accepting `POST` requests compatible with [AWS Firehose HTTP Specifications](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html).

> - `static_configs_labels` - query parameter can be used to set extra static labels. It should contain base64 encoded YAML with labels values. For example:

Reviewer: ```suggestion

You can use the `static_configs_labels` query parameter to set extra static labels. It must contain base64 encoded YAML with labels values.
```

This isn't an endpoint... it's a query param. So it is probably better to add it as it's own sentence vs grouping it with the endpoint description.

>     Generate base64 encoded string from the labels list:

Reviewer: ```suggestion
    For example, you can generate a base64 encoded string from the labels list:
```

    ```bash
    cat << EOF | base64
    ---
    label1: "value1"
    label2: "value2"
    EOF
    LS0tCmxhYmVsMTogInZhbHVlMSIKbGFiZWwyOiAidmFsdWUyIgo=
    ```
  

>     Add query parameter to the url `/awsfirehose/api/v1/push?static_configs_labels=LS0tCmxhYmVsMTogInZhbHVlMSIKbGFiZWwyOiAidmFsdWUyIgo=`

Reviewer: ```suggestion
    Add the query parameter with the base64 encoded value to the URL `/awsfirehose/api/v1/push?static_configs_labels=LS0tCmxhYmVsMTogInZhbHVlMSIKbGFiZWwyOiAidmFsdWUyIgo=`
```

- `/awsfirehose/api/v1/push` - accepting `POST` requests compatible with [AWS Firehose HTTP Specifications](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html).

> You can use the [X-Amz-Firehose-Common-Attributes](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html) header to set extra static labels. Label names should be prefixed with `lbl_`. 
> Label names and label values should be compatible with a [Prometheus data model](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels) specification.

Reviewer: ```suggestion
You can use the [X-Amz-Firehose-Common-Attributes](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html) header to set extra static labels.
Label names must be prefixed with `lbl_`. 
Label names and label values must be compatible with the [Prometheus data model](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels) specification.
```

- `/awsfirehose/api/v1/push` - accepting `POST` requests compatible with [AWS Firehose HTTP Specifications](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html).

> You can use the [X-Amz-Firehose-Common-Attributes](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html) header to set extra static labels. The header can be configured in the **Parameters** section of the Amazon Data Firehose delivery stream configuration. 

Reviewer: ```suggestion
You can use the [X-Amz-Firehose-Common-Attributes](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html) header to set extra static labels. 
You can configure the header in the **Parameters** section of the Amazon Data Firehose delivery stream configuration. 
```

Label names must be prefixed with `lbl_`. 
Label names and label values must be compatible with the [Prometheus data model](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels) specification.

> Example of the valid `X-Amz-Firehose-Common-Attributes` value with two custom labels:

Reviewer 2: When I read this, it wasn't immediately apparent whether these labels are set on the component side or on the AWS firehose side and the component is just reading them out.

Can we make it more explicit and explain where these labels will be applied to (iiuc the read log lines)?
